prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -230,16 +230,13 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 
 	assert(rx_iov_index < rxe->iov_count);
 	assert(rx_iov_offset < rxe->iov[rx_iov_index].iov_len);
+	assert(ep->base_ep.domain->mr_local == 1);
 
-	efa_rdm_ope_try_fill_desc(rxe, rx_iov_index, FI_RECV);
 	if (!rxe->desc[rx_iov_index]) {
-		/* efa_rdm_ope_try_fill_desc() did not fill the desc,
-		 * which means memory registration failed.
-		 * return -FI_EAGAIN here will cause user to run progress
-		 * engine, which will cause some memory registration
-		 * in MR cache to be released.
-		 */
-		err = -FI_EAGAIN;
+		err = -FI_EOPNOTSUPP;
```

Comment:
I think the error code should be `-FI_EINVAL` (invalid argument) as application should provide a valid `desc` in `fi_recv` call.